deno/lib/types.ts

```diff
@@ -1986,7 +1986,7 @@ export interface ZodArrayDef<T extends ZodTypeAny = ZodTypeAny>
   maxLength: { value: number; message?: string } | null;
   uniqueness: {
     identifier?: ArrayUniqueIdentifier;
-    message?: string | ((duplicateElements: Array<unknown>) => string);
+    message?: string | ArrayMessageFunction;
```

Comment:
Why don't you use `T` here?
```suggestion
    message?: string | ArrayMessageFunction<T>;
```